geode-core/src/main/java/org/apache/geode/internal/cache/ha/HARegionQueue.java

```diff
@@ -3799,12 +3799,18 @@ public RemovedEventInfo(Long counter, String regionName, Object key) {
 
     private static volatile ExpirationAttributes testExpAtts;
 
+    private final int expTime;
+
+    ThreadIdentifierCustomExpiry() {
+      expTime = calculateThreadIdExpiryTime();
+    }
+
     @Override
     public ExpirationAttributes getExpiry(Region.Entry entry) {
       // Use key to determine expiration.
       Object key = entry.getKey();
       if (key instanceof ThreadIdentifier) {
-        final int expTime = calculateThreadIdExpiryTime();
+        // TODO: inject subclass of ThreadIdentifierCustomExpiry and move next block to test class
```

Comment:
This requires someone to modify HARegionQueue to provide a mechanism for 1) factory to create ThreadIdentifierCustomExpiry and 2) way to inject the factory into a full-stack Geode for use in integration testing. I don't see a quick way to do this, so I'm deferring that to someone who will be working on HARegionQueue in greater depth.